### DISCUSSION TITLE
Noob: Save / Restore All Desktops

### DISCUSSION THREAD

**Discussion by mark-jerde** [rxhanson/RectanglePro-Community#532] [Q&A] [answered]
Sorry if this is a FAQ I missed.

Retired from Windows development, I now just blog on a MacBook Pro with two external monitors. For blogging I have a total of 21 windows (mostly Chrome and Finder) scattered on 9 desktops, 3 on each monitor. My issue is sometimes the monitors "burp" during login and the windows are not located on the desired desktop. If both monitors "burp" sometimes all 21 windows are on Desktop 1.

I'm looking for an app to quickly / easily save and restore all windows on all desktops. Is this possible with RectanglePro?  (I wrote a few simple scripts on XENIX back in the 1980's, but that was a loooong time ago.) Thx.

**Comment by rxhanson** [comment, marked as answer]
Unfortunately, macOS doesn't provide APIs for managing windows across other desktops/spaces, so the save/restore layout functionality in Rectangle Pro is limited to the active spaces. (On a similar note, the move to next/prev space keyboard shortcut in Rectangle Pro relies on a very specific UI behavior that isn't applicable for any space other than the next/prev one).

**Reply by mark-jerde** [reply]
Ok, thanks.

**Comment by mark-jerde** [comment]
I bought a license anyway. "Long Throw" configured for 6ths is wonderful for my large monitor. 😊  

I don't expect to need to use "Restore Layout" bc I simplified my blogging windows somewhat, and "Window Throw" works quickly on misplaced windows. But here are results of some of my testing FYI.


- All windows closed
- Shutdown (Reopen windows NOT checked)
- Power on, logged on, exported config -01
- Hotkeyed Layout 1 
- Expected 4 Finder windows, just one displayed
- Saved current layout Result One
- Hotkeyed Layout 2
- Expected 2 Finder windows, just one displayed 
- Expected 4 Chrome windows, just one displayed 
- Saved current layout Result One n Two
- Exported config -02-After

[RectangleProConfig-01.json](https://github.com/user-attachments/files/16400575/RectangleProConfig-01.json)
[RectangleProConfig-02-After.json](https://github.com/user-attachments/files/16400576/RectangleProConfig-02-After.json)

**Reply by rxhanson** [reply]
Thanks for purchasing! 

It sounds like perhaps your expectation that restoring a layout is going to automatically open windows for you, whereas it is only launching the app for you. Opening windows for you could be going down a path of automating application behavior, like navigating finder locations or browser addresses (which could vary widely across different apps). I really like this idea so might pursue this at some point, but I'll have to give it more thought.
